examples/lib/annotations.rego

```diff
@@ -0,0 +1,38 @@
+package lib.annotations
+# Annotations helper library used to extract policy metadata information from annotation fields
+# Usage:
+#
+# import data.lib.annotations 
+# annotations = annotations.policy_annotations()
+# 
+
+import data.lib.core
+import data.lib.pods
+
+default paramsAnnotationField = "konstraint.policies/parameters"
+default ignoreAnnotationField = "konstraint.policies/ignore"
+
+get_default(object, field, _default) = output {
+  core.has_field(object, field)
+  object[field] != null
+  output = object[field]
+}
+
+get_default(object, field, _default) = output {
+  core.has_field(object, field) == false
+  output = _default
+}
+
+policy_annotations = return {
+  return := object.union(
+    get_default(core.resource.metadata, "annotations", {}), 
+    get_default(pods.pod.metadata, "annotations", {}))
+} else = return {
+  return :=  get_default(pods.pod.metadata, "annotations", {})
+  return != null
+} else = return {
+  return :=  get_default(core.resource.metadata, "annotations", {})
+  return != null
+} else = return {
+  return := {}
+}
```

Comment:
Thanks for the observations.. I am trying to follow that same logic here, returning defaults from single union and get, but I did not actually get expected result, always had some kind of exception if the "annotations" field is missing. 
Will try to revisit this as it does need some refactoring.

Reply:
OK, so with the new function:
```
all_annotations = return {
	return := object.union(object.get(core.resource.metadata, "annotations", {}), object.get(pods.pod.metadata, "annotations", {}))
}
```
I get this error: ` Error: running test: get test result: running deny rules: run query: run query: evaluating policy: policies/lib/annotations.rego:20: eval_type_error: object.union: operand 1 must be object but got null`

With the previous construct (which acted like a fail-safe switch case... pardon the slang, I'm fairly new to Rego), I actually managed to extract the annotations from any structure of the  arrived Review Object and prioritised them as follows:  `MERGED ANNOTATIONS > POD TEMPLATE SPEC ANNOTATIONS > RESOURCE ANNOTATIONS`.

Reply:
I haven't had a chance to test it with lib.core and lib.pods, but a quick test on the Playground shows it should work as I thought, and swapping in lib.core and lib.pods shouldn't make any difference https://play.openpolicyagent.org/p/gj28LqgnWs. I should have more time to take a closer look later this week.

Reply:
I am testing against hundreds of Yaml resources, some don't have annotations, some don't have metadata field. The function works as expected, but in my case, input resources are given from templating stages and they don't always come back structured correctly.

Reply:
OK, so your issue likely lies in the fact that the `metadata` field may not exist, which `object.get()` returns undefined for. A more idiomatic way to approach this in Rego rather than chaining the `else` statements is to set default values. By doing so, we can actually avoid the `object.get()` calls altogether. See https://play.openpolicyagent.org/p/yctOvtSUzZ

Reply:
oh, thanks for pointing that out.
It seems to a good practice to set defaults for variables, I will try to see how it works out.